playback/src/audio_backend/pipe.rs

```diff
@@ -6,31 +6,57 @@ use std::fs::OpenOptions;
 use std::io::{self, Write};
 
 pub struct StdoutSink {
-    output: Box<dyn Write>,
+    output: Option<Box<dyn Write>>,
+    path: Option<String>,
     format: AudioFormat,
 }
 
 impl Open for StdoutSink {
     fn open(path: Option<String>, format: AudioFormat) -> Self {
         info!("Using pipe sink with format: {:?}", format);
-
-        let output: Box<dyn Write> = match path {
-            Some(path) => Box::new(OpenOptions::new().write(true).open(path).unwrap()),
-            _ => Box::new(io::stdout()),
-        };
-
-        Self { output, format }
+        Self {
+            output: None,
+            path: path,
+            format,
+        }
     }
 }
 
 impl Sink for StdoutSink {
+    fn start(&mut self) -> io::Result<()> {
+        if self.output.is_none() {
+            let output: Box<dyn Write> = match self.path.as_deref() {
+                Some(path) => {
+                    let open_op = OpenOptions::new()
+                        .write(true)
+                        .open(path)
+                        .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
+                    Box::new(open_op)
+                }
+                None => Box::new(io::stdout()),
+            };
+
+            self.output = Some(output);
```

Comment:
I guess that opening the output here, also opens the possibility of closing it on a `stop` event using `drop`.

Reply:
Would ```self.output = None``` accomplish that?

Reply:
I'm not so sure what the value of opening an output and dropping it on stop only to have to recreate and reopen it at start again at a later time? It's not like the output will change in the lifetime of the current instance of librespot?

Reply:
Plus if you dropped it you would think it would break the pipe?

Reply:
> Would `self.output = None` accomplish that?

I think so yes, it'll go out of scope and get cleaned up.

> I'm not so sure what the value of opening an output and dropping it on stop only to have to recreate and reopen it at start again at a later time? It's not like the output will change in the lifetime of the current instance of librespot?

Honestly I am thinking the same thing. I'm not entirely sure how the receiving end will react.

By the same logic it doesn't make much sense to only open the pipe on `start` and not on `open`, because when `librespot` is launched without playback, well, I guess that's the same state as stopping the pipe.

Reply:
From what I understand of `drop` something is automatically dropped when it goes out of scope. Since output is a part of self it can not be dropped until self goes out of scope? In any event you can't explicitly do `drop(self.output)` I tried. The only way that I can see to make output drop is to make it local to `write_bytes` but then you end up recreating it with every write which would be ridiculous I would think?

Reply:
> Honestly I am thinking the same thing. I'm not entirely sure how the receiving end will react.
> 
> By the same logic it doesn't make much sense to only open the pipe on `start` and not on `open`, because when `librespot` is launched without playback, well, I guess that's the same state as stopping the pipe.

The only real reason is that `open` can't currently return errors,lol!!!

Reply:
So yes or no on the `self.output = None` thing? I mean this PR doesn't really change the current behavior that much. Basically all it does is delay the opening of the output. I'm not really a fan of breaking things,lol!!!

Reply:
> well, I guess that's the same state as stopping the pipe.

`aplay` is more than happy to just hangout with `librespot --backend pipe | aplay -q -f cd`. A fifo wouldn't care I'd imagine.

Reply:
You could also try `pacat` to double-check.

Reply:
> You could also try `pacat` to double-check.

Yep `pacat` is perfectly fine to just hangout too. I used `librespot --backend pipe | pacat` and let it sit for a good 20 min before I actually started playback (so the pipe wasn't actually open yet for that 20 min) and it worked as expected.